util/geosop/GeomFunction.cpp

```diff
@@ -46,144 +46,160 @@ using geos::operation::overlayng::OverlayNG;
 /* static private */
 std::map<std::string, GeomFunction*> GeomFunction::registry;
 
-static std::unique_ptr<const PreparedGeometry> prepGeomCache;
-static Geometry *cacheKey;
+class PreparedGeometryCache {
```

Comment:
This is a nice addition.  Shall I add it independently, or do you want to make a separate PR?

Reply:
Probably best for you to proceed independently.

Reply:
Done- thanks.